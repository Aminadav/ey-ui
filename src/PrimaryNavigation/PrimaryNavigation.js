import React, { useState } from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import isEmpty from 'lodash/isEmpty';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import AccountCircle from '@material-ui/icons/AccountCircle';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import withStyles from '@material-ui/core/styles/withStyles';
import Typography from '@material-ui/core/Typography';
import { logout as logoutAction } from '../redux/authentication';

const PrimaryNavigation = ({ user, logout, classes }) => {
  const [open, setOpen] = useState(false);
  const [menuAnchor, setMenuAnchor] = useState();

  return (
    <React.Fragment>
      {user ? (
        <React.Fragment>
          {window.config.mainNav.map(nav => (
            <Button
              key={nav.href}
              component="a"
              href={nav.href}
              className={classes.button}
              data-testid="partner-button"
              target="_blank"
            >
              {nav.label}
            </Button>
          ))}

          <IconButton
            aria-owns={open ? 'menu-appbar' : undefined}
            aria-haspopup="true"
            onClick={() => setOpen(true)}
            color="inherit"
            buttonRef={setMenuAnchor}
            data-testid="user-button"
          >
            <AccountCircle />

            <Typography variant="body1" color="inherit" className={classes.userName}>
              {user.email}
            </Typography>
          </IconButton>

          <Menu
            anchorEl={menuAnchor}
            id="menu-appbar"
            open={open}
            onClick={() => setOpen(false)}
            getContentAnchorEl={null}
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'center',
            }}
            transformOrigin={{
              vertical: 'top',
              horizontal: 'center',
            }}
          >
            <MenuItem disabled>My Organization</MenuItem>
            <MenuItem onClick={logout}>Log Out</MenuItem>
          </Menu>
        </React.Fragment>
      ) : (
        <Button component={Link} to="/login" className={classes.button}>
          Login
        </Button>
      )}
    </React.Fragment>
  );
};

PrimaryNavigation.propTypes = {
  user: PropTypes.shape({}),
  logout: PropTypes.func.isRequired,
  classes: PropTypes.shape({
    button: PropTypes.string.isRequired,
    userName: PropTypes.string.isRequired,
  }).isRequired,
};

PrimaryNavigation.defaultProps = {
  user: null,
};

const styles = theme => ({
  button: {
    color: theme.palette.primary.contrastText,
    paddingLeft: '1rem',
    paddingRight: '1rem',
  },
  userName: {
    marginLeft: '.5em',
  },
});

const mapStateToProps = state => ({
  user: isEmpty(state.user) ? null : state.user,
});

export default compose(
  withStyles(styles),
  connect(
    mapStateToProps,
    { logout: logoutAction },
  ),
)(PrimaryNavigation);
