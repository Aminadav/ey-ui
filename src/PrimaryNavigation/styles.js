import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(theme => ({
  button: {
    color: theme.palette.primary.contrastText,
    paddingLeft: '1rem',
    paddingRight: '1rem',
  },
  userName: {
    marginLeft: '.5em',
  },
}));

export default useStyles;
