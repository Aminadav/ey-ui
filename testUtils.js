import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
import ThemeProvider from '@material-ui/styles/ThemeProvider';

/**
 * Render dom wrapped with material-ui's ThemeProvider.
 */
export function renderWithTheme(ui) {
  return {
    ...render(<ThemeProvider theme={createMuiTheme()}>{ui}</ThemeProvider>),
  };
}

/**
 * Render dom wrapped with multiple dependent context providers.
 */
export function renderWithProviders(ui) {
  return {
    ...render(
      <MemoryRouter initialEntries={['/']}>
        <ThemeProvider theme={createMuiTheme()}>{ui}</ThemeProvider>
      </MemoryRouter>,
    ),
  };
}

export default { renderWithTheme };
