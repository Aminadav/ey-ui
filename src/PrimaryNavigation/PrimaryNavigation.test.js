import React from 'react';
import { renderWithProviders } from '../../testUtils';
import PrimaryNavigation from './PrimaryNavigation';

describe('PrimaryNavigation', () => {
  it('matches the snapshot', () => {
    const { container } = renderWithProviders(<PrimaryNavigation {...{ onLogout() {} }} />);
    expect(container).toMatchSnapshot();
  });

  it('should render nav buttons if user is passed in', () => {
    const { getByTestId } = renderWithProviders(
      <PrimaryNavigation
        {...{
          user: { email: 'foo', name: 'foo' },
          mainNav: [{ label: 'Partners', href: 'http://partner' }],
          onLogout() {},
        }}
      />,
    );
    expect(getByTestId('partner-button')).toBeInTheDocument();
  });

  // TODO: test that clicking the user button pops open the user nav
});
