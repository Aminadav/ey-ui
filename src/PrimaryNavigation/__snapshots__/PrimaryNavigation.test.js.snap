// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryNavigation matches the snapshot 1`] = `
<div>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root MuiButton-root makeStyles-button-1 MuiButton-text"
    href="/login"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    >
      Login
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
</div>
`;
